bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -707,11 +707,11 @@ public String toString() {
 
         private @Nullable State calculateDelta() {
             if (newState instanceof QuantityType newStateQuantity) {
-                QuantityType result = newStateQuantity.subtract((QuantityType) acceptedState);
+                QuantityType result = newStateQuantity.subtract((QuantityType) acceptedState.get());
```

Comment:
I don't like this unprotected cast here. But don't worry too much because #18144 does already override that.

Reply:
It is already checked against empty in `alwaysAccept()`

Reply:
I know. But my grandma told me never to write Java code like that..

Reply:
Wise grandma! I've added empty checks in the corresponding methods.